app/partials/app.cjsx

```diff
@@ -3,15 +3,81 @@ auth = require 'panoptes-client/lib/auth'
 IOStatus = require './io-status'
 MainHeader = require './main-header'
 MainFooter = require './main-footer'
+GeordiClient = require 'zooniverse-geordi-client'
 {generateSessionID} = require '../lib/session'
 
+class GeordiLogger
+  constructor: (@state, @geordi) ->
+
+  keys: {}
+
+  geordi: null
+
+  instance: =>
+    @geordi = @geordi || @makeGeordi @keys?.projectToken
+
+  makeGeordi: (projectSlug) ->
+    new GeordiClient
+      server: @state?.env
+      projectToken: projectSlug || ''
+      zooUserIDGetter: () => @state.user?.id
+      subjectGetter: () => @keys?.subjectID
+
+  makeHandler: (defType) ->
+    instance = @instance
+    (eventData, eventType) ->
+        eventType = defType if typeof linkType isnt 'string'
```

Comment:
`linkType` isn't defined in this method. Should this be `eventType`? What happens here if `defType` isn't a string either?